web-service/helpers.py

```diff
@@ -37,9 +37,14 @@
 
 ################################################################################
 
-def BuildResponse(objects):
+def BuildResponse(objects, request_headers):
     metadata_output = []
 
+    if 'Origin' in request_headers:
+        base_url = request_headers['Origin']
+    else:
+        base_url = 'https://' + app_identity.get_application_id() + '.appspot.com'
```

Comment:
Nit: can we make this a constant up top?  You have a separate FR to change the base_url, so it will be good to have only place to adjust this